CodeAnalysis/Xamarin.Interactive.CodeAnalysis/CodeAnalysis/Evaluating/EvaluationService.cs

```diff
@@ -383,6 +383,9 @@ public Task EvaluateAllAsync (CancellationToken cancellationToken = default)
 
                 events.Observers.OnNext (lastCellFinishedEvent);
 
+                if (status != EvaluationStatus.Success)
+                    NotifyEvaluationComplete (evaluatableCodeCell.CodeCellId, status);
```

Comment:
`Success`-ful evals will be pushed into NotifyEvaluationComplete once they've actually been evaluated and represented via the message channel (for normal agents) or via the SignalR hub for WASM. This catches non-Success evals, which otherwise just end up spinning forever.